### ISSUE TITLE
Invalid credentials when running on AWS

### ISSUE THREAD

**Issue by RhinocerosBomb** [tlaplus/tlaplus#520]
Tried on Windows 10 and Ubuntu 20. Set the environment variables to have `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` and I have made sure that they are correct, but I am getting this error when I run: 

```
Invalid credentials, please check the environment variables (AWS_ACCESS_KEY_ID & AWS_SECRET_ACCESS_KEY) are correctly set up and picked up by the Toolbox.

Please visit the Toolbox help and read section 4 of "Cloud based distributed TLC" on how to setup authentication.
```

**Comment by lemmy** [comment]
Where have you set the environment variables and how do you launch the Toolbox?

**Comment by RhinocerosBomb** [comment]
The environment variables are set using command prompt/line

On Windows:
```
set AWS_ACCESS_KEY_ID =XXX
set AWS_SECRET_ACCESS_KEY=XXX
```

On Linux:
```
export AWS_ACCESS_KEY_ID =XXX
export AWS_SECRET_ACCESS_KEY=XXX
```

The way I start the toolbox is by clicking on the toolbox.exe file from the downloaded package

**Comment by lemmy** [comment]
Can you try launching the Toolbox from the shell/command in which you set the environment vars.

**Comment by RhinocerosBomb** [comment]
Works! I do get an error with vCPU limit on AWS. What type of instance does it try to spin up?

**Comment by lemmy** [comment]
Unless it's overridden by a property... https://github.com/tlaplus/tlaplus/blob/859203e2673c133f3ef6b652bc37673b5d08d891/toolbox/org.lamport.tla.toolbox.jclouds/src/org/lamport/tla/toolbox/jcloud/EC2CloudTLCInstanceParameters.java#L148-L156

**Comment by RhinocerosBomb** [comment]
So, when I create a new Java file overriding that function, do I have to rebuild toolbox or is there a convenient way to override per model run?

**Comment by lemmy** [comment]
No, you don't create a new Java file but simply launch the Toolbox (see toolbox.ini) with `-Daws-ec2.instanceType=yourFavInstType`.  A different instance type might not work with the provisioning scripts of CloudTLC, though.

**Comment by RhinocerosBomb** [comment]
OK. Thanks for your help!

**Comment by zitro3** [comment]
> Works! I do get an error with vCPU limit on AWS. What type of instance does it try to spin up?

I was trying this before on Azure and AWS but was not successful due to some other reason. I encountered that error before and the solution was that you have to request AWS to increase your vCPU quota thru your subscription on that specific region where VMs will be instantiated or launched for that specific VMs (AWS Instance Type). AWS will review and wait if they can give you new limit, within a day in my case. I hope this helps.
